pikitlib.py

```diff
@@ -16,70 +34,203 @@ def __init__(self, id):
         # get and save button state
         
         # A-0,B-1,X-2,Y-3
-        self.lastButtonValues = []
-        self.lastButtonValues.append(self.nt.getBoolean("Button0", False))
-        self.lastButtonValues.append(self.nt.getBoolean("Button1", False))
-        self.lastButtonValues.append(self.nt.getBoolean("Button2", False))
-        self.lastButtonValues.append(self.nt.getBoolean("Button3", False))
 
+        # A, B, X, Y, L bumpter, R bumber, back, start, big shinny button
```

Comment:
Why can't we read the button array with a single read from the network tables?   Then we'll know how many buttons there are, and we will know the value of all of the buttons in the self.newButtonValues[] array.
Same for reading the axis values (as an array).